prusti-contracts/prusti-contracts/src/core_spec/default.rs

```diff
@@ -0,0 +1,45 @@
+use crate::*;
+
+#[extern_spec]
+trait Default {
+    #[refine_spec(where Self: Copy + PureDefault, [pure])]
+    fn default() -> Self;
+}
+
+/// Specifies that `Default::default`, if implemented, is a pure method, allowing its usage in specs.
+///
+/// Does not apply to types that do not implement `Copy`, since pure methods can only involve `Copy` types.
+pub auto trait PureDefault {}
```

Comment:
I don't know if I asked before: can `PureDefault` be declared with `Copy` as a super-trait? Or does that not work for auto traits?

Reply:
They unfortunately cannot have super traits, I just checked again to make sure. The explanation sounds like it's not planned either, rather a principle I'd disagree with.

<img width="468" alt="image" src="https://user-images.githubusercontent.com/827762/212780466-39f8b495-736f-4872-92f4-efd0984d311b.png">